core/src/main/java/org/apache/calcite/config/CalciteSystemProperty.java

```diff
@@ -420,6 +420,18 @@ public final class CalciteSystemProperty<T> {
   public static final CalciteSystemProperty<Integer> FUNCTION_LEVEL_CACHE_MAX_SIZE =
       intProperty("calcite.function.cache.maxSize", 0, v -> v >= 0);
 
+  /**
+   * The length of code before automatic method splitting is enabled
+   *
+   * <p>A value of <code>0</code>, the default, disables automatic method splitting.
+   *
+   * <p>Some queries can generate methods exceeding the JVM limit of 4000 characters per method.
+   * Use this setting to automatically detect and split methods larger than the specified
+   * limit.
+   */
+  public static final CalciteSystemProperty<Integer> METHOD_SPLITTING_THRESHOLD =
```

Comment:
nit: proposal (to make the new property robust in case of negative values)
```
public static final CalciteSystemProperty<Integer> METHOD_SPLITTING_THRESHOLD =
      intProperty("calcite.linq.method_splitting_threshold", 0, v -> v >= 0);
```